packages/gatsby/cache-dir/component-renderer.js

```diff
@@ -61,7 +61,10 @@ class ComponentRenderer extends React.Component {
     // This is only useful on delayed transitions as the page will get rendered
     // without the necessary page resources and then re-render once those come in.
     emitter.on(`onPostLoadPageResources`, e => {
-      if (e.page.path === loader.getPage(this.state.location.pathname).path) {
+      if (
+        loader.getPage(this.state.location.pathname) &&
```

Comment:
This could possibly be that the page resource might not exist. Putting a checking to guard from having to encounter error at this part.